CAAppKitBridge/Source/NSView+CAMethods.m

```diff
@@ -0,0 +1,148 @@
+/* NSView+CAMethods.m
+
+   Copyright (C) 2018 Free Software Foundation, Inc.
+
+   Author: Stjepan Brkic <stjepanbrkicc@gmail.com>
+   Date: June 2018
+
+   This file is part of QuartzCore/CAAppKitBridge.
+
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2 of the License, or (at your option) any later version.
+
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; see the file COPYING.LIB.
+   If not, see <http://www.gnu.org/licenses/> or write to the
+   Free Software Foundation, 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110-1301, USA.
+*/
+
+#import "GSCAData.h"
+#import "NSView+CAMethods.h"
+
+@implementation NSView (NSViewCAmethods)
+- (BOOL) wantsLayer
+{
+  if (self->_coreAnimationData == nil)
+    {
+      return NO;
+    }
+  GSCAData * GSCAData = self->_coreAnimationData;
+  return GSCAData->_wantsLayer;
+}
+
+- (void) setWantsLayer:(BOOL)newValue
+  {
+  if (newValue == NO)
+    {
+      return;
```

Comment:
If value is `NO`, you should check if `self` is root of the layer-wanting tree. If so, it should remove the layers on `self` and children, and also remove the renderer and GL context from `self`.

You can do that in a future commit.